File: Rightside view BFS.java
// Time complexity 0(n)
// space complexity 0(n) [size of the Queue, auxiliary space ]
// Code successfully executed in leetcode.
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        //base case 
        List<Integer> result = new ArrayList<>();
        if(root == null) return result;
        
        Queue<TreeNode> q1 = new LinkedList<>();
        q1.add(root);
        while(!q1.isEmpty()){
             int size = q1.size();
            // Contains last value of the node
             int temp =0;   
          for(int i=0;i<size;i++){
                  TreeNode node = q1.poll(); 
                  if(i==0){
                      temp = node.val;
                  }
                 if(node.right != null) q1.add(node.right);
                 if(node.left != null) q1.add(node.left);
            }
            result.add(temp);
        }
        return result;
    }
}

File: Right side view BFS.py
# // Time Complexity : O(n)
# // Space Complexity : O(n)
# // Did this code successfully run on Leetcode : Yes
# // Any problem you faced while coding this : No 

# We use DFS level order,  we store the size at each level, after each level we store the last element into the res array


# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: TreeNode) -> List[int]:
        if not root:
            return []
        
        res=[]
        
        q = deque()
        q.append(root)
        
        while q:
            l=[]
            size = len(q)
            for i in range(size):
                x = q.popleft()
                l.append(x.val)
                if x.left : q.append(x.left)
                if x.right: q.append(x.right)
            res.append(x.val)
            
        return (res)
                    